src/components/vm/vmReplaceSpiceDialog.jsx

```diff
@@ -30,17 +36,28 @@ import { fmt_to_fragments } from 'utils.jsx';
 import { NeedsShutdownAlert } from '../common/needsShutdown.jsx';
 import { domainReplaceSpice } from '../../libvirtApi/domain.js';
 
+import './vmReplaceSpiceDialog.scss';
+
 const _ = cockpit.gettext;
 
-export const ReplaceSpiceDialog = ({ vm }) => {
+export const ReplaceSpiceDialog = ({ vm, vms }) => {
+    const spiceVMs = vms?.filter(vm => vm.inactiveXML?.hasSpice);
+    // sort spicVMs by name, put the current VM first
+    spiceVMs?.sort((a, b) => (a.name === vm.name) ? -1 : (b.name === vm.name) ? 1 : a.name.localeCompare(b.name));
+    const isMultiple = spiceVMs?.length > 1;
+
     const Dialogs = useDialogs();
     const [error, dialogErrorSet] = useState(null);
     const [inProgress, setInProgress] = useState(false);
+    const [selected, setSelected] = useState(isMultiple ? [spiceVMs[0].id] : []);
 
     function onReplace() {
         setInProgress(true);
 
+        const selectedVMs = spiceVMs?.filter(vm => selected.includes(vm.id)) || [];
+
         return domainReplaceSpice({ connectionName: vm.connectionName, id: vm.id })
+                .then(() => Promise.allSettled(selectedVMs.map(vm => domainReplaceSpice({ connectionName: vm.connectionName, id: vm.id }))))
```

Comment:
This converts all VMs in parallel, which is fine for Cockpit and pipelining over slow connections, but might be stressing poor libvirt. Would it make sense to be conservative and convert them one-by-one? Or do we know that libvirt is fine with this?

Reply:
The conversion is not that expensive, but let's check. My VM can host about 60 running micro-VMs until it runs out of memory:
```
for i in `seq 60`; do virt-install --memory 50 --pxe --virt-type qemu --os-variant alpinelinux3.8 --disk none --wait 0 --name test$i; done
```
This takes about 5 seconds and succeeds. I'll try again with serial conversion, if that doesn't take an undue amount of time, we can play it safe and do it serially.

Reply:
> This takes about 5 seconds and succeeds.

The conversion takes 5 seconds, right? (Not the creation of the VMs.)

Reply:
Right, I meant the "Replace" button. Creating the VMs takes much longer.